File: 1683-maximum-number-of-coins-you-can-get/maximum-number-of-coins-you-can-get.js
/**
 * @param {number[]} piles
 * @return {number}
 */
var maxCoins = function(piles) {
    piles.sort((a,b) => a-b);

    let itr = piles.length / 3, ans = 0;
    let ptr = piles.length-2;
    while(itr--) {
        ans += piles[ptr];
        ptr -= 2;
    }
    return ans;
};

File: 1683-maximum-number-of-coins-you-can-get/maximum-number-of-coins-you-can-get.java
class Solution {
    public int maxCoins(int[] piles) {
        Arrays.sort(piles);
        int ans = 0, itr = piles.length / 3;
        int ptr = piles.length-2;
        while(itr-- > 0) {
            ans += piles[ptr];
            ptr -= 2;
        }
        return ans;
    }
}